controllers/discordactions.js

```diff
@@ -407,17 +408,30 @@ const setRoleToUsersWith31DaysPlusOnboarding = async (req, res) => {
   }
 };
 
+const generateRandomId = () => {
+  return crypto.randomBytes(16).toString("hex");
+};
+
 const generateInviteForUser = async (req, res) => {
   try {
-    const { userId } = req.query;
+    const { userId, dev } = req.query;
     const userIdForInvite = userId || req.userData.id;
-
-    const modelResponse = await discordRolesModel.getUserDiscordInvite(userIdForInvite);
-
-    if (!modelResponse.notFound) {
-      return res.status(409).json({
-        message: "User invite is already present!",
-      });
+    let inviteLink = "";
+    let randomId = "";
+
+    if (!dev) {
+      const modelResponse = await discordRolesModel.getUserDiscordInvite(userIdForInvite);
+      if (!modelResponse.notFound) {
+        return res.status(409).json({
+          message: "User invite is already present!",
+        });
+      }
+    } else {
+      const modelResponse = await discordRolesModel.getUserDiscordInvite(userIdForInvite);
+      if (!modelResponse.notFound) {
+        // Generate a random ID
+        randomId = generateRandomId();
```

Comment:
Why are you doing this?

Reply:
I have allowed superusers to create the discord invites as many times as the superuser wants. So, for the first time the function will take id from the superuser data and afterwards I am generating the userId and passing it to the db.

Reply:
You are generating random id 
instead every time you can store the super userID

Reply:
I think storing multiple invites with same id is not a good idea.